### ISSUE TITLE
No shell is loaded (hanging while loading CMD, Powershell, WSL)

### ISSUE THREAD

**Issue by corelon** [felixse/FluentTerminal#78]
Whatever shell I set as default, it will hang waiting for it to load.
I then need to close the window and exit the program from task bar.
I have enabled Developer mode (since I have been running WSL for a while now).
Any help would be greatly appreciated.

**Comment by felixse** [comment]
So you do see the main window, but the terminal is empty (just a cursor is displayed)?
Do you have any anti virus installed that might interfere starting the shell process?

**Comment by corelon** [comment]
Hi,

Indeed I see only the cursor. I do have antivirus installed but cannot really tweak it as it is a work machine.

**Comment by felixse** [comment]
I'm afraid there won't be much we can do about that. Do other emulators like Hyper work?

**Comment by corelon** [comment]
Yes. Hyper, Terminus, ElecTerm, ConEmu and Cmder work

**Comment by Riebart** [comment]
While it is hung and waiting, here's a few things I would check:

- Is there a FluentTerminal system tray process running?
- If so, does that process have TCP listening ports up? If so, what are the firewall restrictions on that?
- What endpoint protection are you using?
- What is the output of `CheckNetIsolation.exe LoopbackExempt -s` run as your user?
  - Does it help to run the `CheckNetIsolation.exe` snippet from the README as an administrative user for explicitly permitting the socket connection for this app?

**Comment by corelon** [comment]
Hi,
This is the output of CheckNetIsolation.exe LoopbackExempt -s:
List Loopback Exempted AppContainers

[1] -----------------------------------------------------------------
    Name: microsoft.microsoftedge_8wekyb3d8bbwe
    SID:  S-1-15-2-3624051433-2125758914-1423191267-1740899205-1073925389-3782572162-737981194

[2] -----------------------------------------------------------------
    Name: windows_ie_ac_001
    SID:  S-1-15-2-1430448594-2639229838-973813799-439329657-1197984847-4069167804-1277922394

[3] -----------------------------------------------------------------
    Name: 001
    SID:  S-1-15-2-3624051433-2125758914-1423191267-1740899205-1073925389-3782572162-737981194-4256926629-1688279915-2739229046-3928706915

[4] -----------------------------------------------------------------
    Name: microsoft.microsoftedge_8wekyb3d8bbwe
    SID:  S-1-15-2-3624051433-2125758914-1423191267-1740899205-1073925389-3782572162-737981194

[5] -----------------------------------------------------------------
    Name: AppContainer NOT FOUND
    SID:  S-1-15-2-3232256523-3432637531-61793852-1933176919-3501339092-2872291700-4169625008

OK.

It is the same regardless of if I run this as a user or admin.
The system tray process is indeed running
I am using Symantec Endpoint protection (locked unfortunately)
As for the TCP listening ports could you please give me some guidance?

Thanks in advance.

**Comment by felixse** [comment]
NetIsolation should be no issue here, since we switched to AppServiceConnection for the inter process communication. I think your problem is in fact Symantec that is blocking our call to open the shell process (I can understand why an anti virus software would find that suspicious). The interesting part is why Hyper etc. are working. Can you see anything in Symantec's log (if you can access it)?

**Comment by corelon** [comment]
Apologies for the late reply. Is there anything specific in the log that I should be looking for?

**Comment by corelon** [comment]
How can I uninstall it and try installing it again?

**Comment by felixse** [comment]
I guess everything with FluentTerminal could be interesting. You can uninstall it by right clicking in the start menu (Just like with normal Windows Store Apps)

**Comment by corelon** [comment]
When I try to install again I get the following:
App installation failed with error message: An error occurred while deleting the package's previously existing application data. (0x80073d05)

**Comment by corelon** [comment]
Hi.
Just installed the latest version 0.3.1.0 and I am still getting the same issue of not loading any shell. 
The window opens normally, I can open the settings menu, but I cannot do anything else. In other words it is not "frozen", it is just the shell that is not loading.

**Comment by skipmeister123** [comment]
You know, I have this issue pretty often.  Most of the time, what ends up happening is the system tray process somehow died.  Re-launching usually fixes it for me.  However, it sounds like that might not be the case for you.  Can you check the task manager to see if the process is running?  If so, force quit it.

The other idea I have is maybe a bit far-fetched.  If you've got two machines you use FT on, and both of them aren't set up the same way (eg, one of your machines has WSL and the other does not), that could cause some problems.  Since the settings are in Roaming, they'll be synced.  If you set the default terminal to WSL on one machine but don't have WSL installed on the other, it wouldn't be able to open that default tab.  Like I said - far-fetched, but since I don't know your setup, it's worth considering, no?

--
Hans Kokx
On 1/10/2019 5:17:29 AM, Nikos Tzanos <notifications@github.com> wrote:
Hi.
Just installed the latest version 0.3.1.0 and I am still getting the same issue of not loading any shell.
The window opens normally, I can open the settings menu, but I cannot do anything else. In other words it is not "frozen", it is just the shell that is not loading.

**Comment by corelon** [comment]
Hi,
Thank you very much for the info. I have FT installed on my work laptop and the profile is not synced.
I have WSL installed, but still it is Powershell that is the default shell that is not opening. I have also tried opening tabs with WSL and CMD but still the same issue.
Regarding the tray icon, I saw that when I start FT it starts as well whille it was not there before. If I close FT but not quit the tray icon then it will stay there.

I hope I helped.
